cluster-autoscaler/cloudprovider/aws/aws_wrapper.go

```diff
@@ -107,59 +109,50 @@ func (m *awsWrapper) getAutoscalingGroupsByNames(names []string) ([]*autoscaling
 	return asgs, nil
 }
 
-func (m *awsWrapper) getAutoscalingGroupNamesByTags(kvs map[string]string) ([]string, error) {
-	// DescribeTags does an OR query when multiple filters on different tags are
-	// specified. In other words, DescribeTags returns [asg1, asg1] for keys
-	// [t1, t2] when there's only one asg tagged both t1 and t2.
+func (m *awsWrapper) getAutoScalingGroupsByTags(configs []asgAutoDiscoveryConfig) ([]*autoscaling.Group, error) {
+	if len(configs) == 0 {
+		return nil, nil
+	}
+
 	filters := []*autoscaling.Filter{}
-	for key, value := range kvs {
-		filter := &autoscaling.Filter{
-			Name:   aws.String("key"),
-			Values: []*string{aws.String(key)},
-		}
-		filters = append(filters, filter)
-		if value != "" {
-			filters = append(filters, &autoscaling.Filter{
-				Name:   aws.String("value"),
-				Values: []*string{aws.String(value)},
-			})
+	tagNames := []string{}
+	for _, config := range configs {
+		for key, value := range config.Tags {
+			filter := &autoscaling.Filter{
```

Comment:
i see it's the same as the old code, but it's a little strange that for the key filter you create a temp object `filter` and for the value object you inline it within the append. imo inline it in both cases

Reply:
Good point, I'll update to inline it